### PR TITLE
Opinion bolding

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
- highlights multiple instances of a given word
- highlights all of a given word if the given word is a substring of a a word in the entry. Example: "should" should match "shouldn't" as well as "should".
-
### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Product-Board-30f97e4eb01246dbb8264253fb385073?p=e0c9347454674d3eb698b236ebe672cf

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | not yet
Self-Review: Have you done an initial self-review of the code below on Github? |
Design Review: If applicable, have you compared the coded design to the mockups? | (N/A

### CODE REVIEW

File: services/QuillLMS/client/setEnvVars.js [hunk 1]
```diff
@@ -1 +1 @@
-process.env.GOLANG_FANOUT_URL = 'https://us-central1-comprehension-247816.cloudfunctions.net/comprehension-fanout-staging';
+process.env.GOLANG_FANOUT_URL = 'https://staging.comprehension-feedback.quill.org/';
```
Comment (happythenewsad) [inline services/QuillLMS/client/setEnvVars.js:1]: This change is not directly related to the feature, but is required in order to run the feature locally and should be in place for everyone.

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+class StringFormatting {
```
Comment (happythenewsad) [inline services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: I'd like to move toward expressing static utility functions as class methods. It allows use to put functions in logical groups, and also adds namespacing and signals intent.

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import StringFormatting from "./stringFormatting"
+
+
+describe('stringFormatting', () => {
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.test.ts]: For completeness, should there be a test for `wordsToFormat: string[]`?

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: I like the idea of this extraction.  Just a thought, but what about going a step further and renaming the function name the function name to  `StringFormatting.highlightSpellingGrammar(...)`  since the class is already expressing that this is formatting?

**Review by brendanshean — APPROVED** [review]
Minor comments, but looks good to me.

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts (hunk 2 above)
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: I'm not strictly against it in this context, but I do think that in general, we are moving towards a functional rather than OO approach in our Javascript and it probably doesn't make sense to backtrack on that here. I think file structure can group things logically without the use of classes.

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  static highlightSpellingGrammar = (str: string, wordsToFormat: string | string[]) => {
+    let wordArray = [].concat(wordsToFormat)
+    let newString = str
+    wordArray.forEach((word) => {
+      var matched = newString.match(`${word}[\\w\']*`)
+      var augmentedRegex = new RegExp(`${matched[0]}`, 'g')
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: Use `const` instead of `var` here: https://airbnb.io/javascript/#references--prefer-const

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts (hunk 2 above)
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: Also, I think that since this function is used in such a specific capacity, it's better to keep it closer in the file structure to where it is actually employed (in only one place, right now) than all the way out in `Shared/utils`. Curious if @eadams17 has thoughts about either of these points.

**Review by emilia-friedberg — CHANGES_REQUESTED** [review]
Small change requested around a style guide issue, two more points for your consideration.

File: services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts]: I agree with @emilia-friedberg. In general, React is moving away from the use of classes and is encouraging functional development with its newer features so I think we want to continue on that course. I also agree on the second point; there is already a `/libs` directory within the Comprehension app where I think makes more sense to house this since it's specific to Comprehension

**Review by emilia-friedberg — APPROVED** [review]
LGTM!
